### ISSUE TITLE
Route data maps ignore meta-merge options in 0.7.0, breaking compatibility

### ISSUE THREAD

**Issue by svdm** [metosin/reitit#679]
In 0.6.0, nested map data is merged in a way that matches meta-merge, supporting `^:replace`:
```clojure
user> (meta-merge {:roles {:foo false :bar true}} {:roles ^:replace {:foo true}})
{:roles {:foo true}}

user> (r/routes (r/router
                 ["/context" {:roles {:foo false :bar true}}
                  ["/leaf" {:roles ^:replace {:foo true}}]]))
[["/context/leaf" {:roles {:foo true}}]]
```

In 0.7.0, this is no longer the case, `^:replace` is not respected:
```clojure
user> (r/routes (r/router
                 ["/context" {:roles {:foo false :bar true}}
                  ["/leaf" {:roles ^:replace {:foo true}}]]))
[["/context/leaf" {:roles {:foo true, :bar true}}]]
```

This is not documented in the changelog so I assume it's not intentional.

**Comment by svdm** [comment]
I have found a workaround of sorts by fully disabling `:update-paths` functionality:
```clojure
user> (r/routes (r/router
                             ["/context" {:roles {:foo false :bar true}}
                              ["/leaf" {:roles ^:replace {:foo true}}]]
                             {:update-paths nil}))
[["/context/leaf" {:roles {:foo true}}]]
```
As a result presumably losing out on the schema merging fixes.

The root cause is that `update-paths` strips metadata from nested maps (strictly: rebuilds the maps from scratch without retaining metadata), preventing meta-merge from ever seeing it, so it's not surprising that meta-merge directives on maps have no effect.

**Comment by ikitommi** [comment]
You are right, this is a regression.

**Comment by ikitommi** [comment]
fixed and released in 0.7.1.
